src/gap.c

```diff
@@ -1608,37 +1610,33 @@ void InitializeGap (
                 }
             }
         }
-    }
 #endif
 
-    /* if we are restoring, load the workspace and call the post restore   */
-    if ( SyRestoring ) {
+        // we are restoring, load the workspace and call the post restore
         ModulesInitModuleState();
         LoadWorkspace(SyRestoring);
         SyRestoring = NULL;
 
         /* Call POST_RESTORE which is a GAP function that now takes control, 
            calls the post restore functions and then runs a GAP session */
-        if (POST_RESTORE != (Obj) 0 &&
-            IS_FUNC(POST_RESTORE)) {
+        if (POST_RESTORE != 0 && IS_FUNC(POST_RESTORE)) {
           TRY_IF_NO_ERROR {
             CALL_0ARGS(POST_RESTORE);
           }
         }
-    }
 
+        return;
+    }
+#endif // GAP_ENABLE_SAVELOAD
 
     /* otherwise call library initialisation                               */
-    else {
```

Comment:
Note that it is safe to remove this `else {` because we added a `return;` a few lines above; and that in turn is OK because `SyLoadSystemInitFile` is always zero when `SyRestoring` is non-zero; so the only thing we "loose" is the call to `ModulesCheckInit();`. While it would have been trivial enough to add this before the removed `return`, I chose not to, because it just performs some sanity checks, which are pointless to run *after* we finished execution (i.e., after `POST_RESTORE` returns). Besides, those checks are mostly meant to catch errors in constant internal tables, so any errors it detects would be caught by us (as they'd show up on every launch) and thus not included in a released GAP version.